ckanext/xloader/utils.py

```diff
@@ -9,6 +9,36 @@
 from decimal import Decimal
 
 import ckan.plugins as p
+from ckan.plugins.toolkit import config
+
+# resource.formats accepted by ckanext-xloader. Must be lowercase here.
+DEFAULT_FORMATS = [
+    "csv",
+    "application/csv",
+    "xls",
+    "xlsx",
+    "tsv",
+    "application/vnd.ms-excel",
+    "application/vnd.openxmlformats-officedocument.spreadsheetml.sheet",
+    "ods",
+    "application/vnd.oasis.opendocument.spreadsheet",
+]
+
+
+class XLoaderFormats(object):
+    formats = None
+
+    @classmethod
+    def is_it_an_xloader_format(cls, format_):
+        if cls.formats is None:
+            cls._formats = config.get("ckanext.xloader.formats")
+            if cls._formats is not None:
```

Comment:
I assume we don't just use `if cls._formats:` because if the config is an empty list, we want to preserve that. Should there be a comment to explain it?

Reply:
@ThrawnCA yeah I believe so, I have tested with an empty list and this seems to work.

I did not write this class, just moved it from helpers to utils a while ago. But added in the in-line comment now